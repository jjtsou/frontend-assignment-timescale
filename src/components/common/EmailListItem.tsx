import { Text } from '@chakra-ui/react';
import { memo } from 'react';

type EmailListItemTypes = {
  isSelected?: boolean;
  onClick?: () => void;
  value: string;
};

const EmailListItem = ({
  isSelected = false,
  onClick,
  value,
}: EmailListItemTypes): JSX.Element => {
  const shouldHavePointerEvents = !!onClick;
  return (
    <Text
      cursor={shouldHavePointerEvents ? 'pointer' : 'initial'}
      color={isSelected ? 'blackAlpha' : 'gray.500'}
      fontWeight={'semibold'}
      onClick={onClick}
      _hover={shouldHavePointerEvents ? { bg: 'gray.200' } : {}}
      py={1}
      px={2}
      rounded={4}
    >
      {value}
    </Text>
  );
};

export default memo(EmailListItem);
