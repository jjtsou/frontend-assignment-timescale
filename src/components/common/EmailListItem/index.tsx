import { Button, Text } from '@chakra-ui/react';
import { memo } from 'react';

type EmailListItemTypes = {
  isSelected?: boolean;
  onClick?: () => void;
  value: string;
};

const EmailListItem = ({
  isSelected = false,
  onClick,
  value,
}: EmailListItemTypes): JSX.Element => {
  return (
    <Text
      as={onClick ? Button : Text}
      color={isSelected ? 'blackAlpha' : 'gray.500'}
      fontWeight={'semibold'}
      onClick={onClick}
      py={1}
      px={2}
    >
      {value}
    </Text>
  );
};

export default memo(EmailListItem);
