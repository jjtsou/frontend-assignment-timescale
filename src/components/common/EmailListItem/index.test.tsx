import { render } from '@testing-library/react';
import EmailListItem from '../EmailListItem';

describe('EmailListItem', () => {
  it('should render correctly', () => {
    const { container } = render(<EmailListItem value="test@example.com" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render as selected when isSelected prop is true', () => {
    const { container } = render(
      <EmailListItem value="test@example.com" isSelected />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render as a button when onClick prop is provided', () => {
    const { container } = render(
      // eslint-disable-next-line @typescript-eslint/no-empty-function
      <EmailListItem value="test@example.com" onClick={() => {}} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
