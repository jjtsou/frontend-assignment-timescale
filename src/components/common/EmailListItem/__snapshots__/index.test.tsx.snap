// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailListItem should render as a button when onClick prop is provided 1`] = `
<button
  class="chakra-button chakra-text css-127jow9"
  type="button"
>
  test@example.com
</button>
`;

exports[`EmailListItem should render as selected when isSelected prop is true 1`] = `
<p
  class="chakra-text chakra-text css-1pldbq1"
>
  test@example.com
</p>
`;

exports[`EmailListItem should render correctly 1`] = `
<p
  class="chakra-text chakra-text css-1k47jjv"
>
  test@example.com
</p>
`;
